lib/matplotlib/inset.py

```diff
@@ -79,6 +79,13 @@ def _shared_setter(self, prop, val):
 
         artist.setp([self._rectangle, *self._connectors], prop, val)
 
+    @artist.Artist.axes.setter
+    def axes(self, new_axes):
+        # Set axes on the rectangle (required for some external transforms to work) as
+        # well as the InsetIndicator artist.
+        self.rectangle.axes = new_axes
+        artist.Artist.axes.fset(self, new_axes)
```

Comment:
I copied the setter logic here from the [artist within artist example](https://matplotlib.org/stable/gallery/text_labels_and_annotations/line_with_text.html).  I would not have figured it out by myself - thanks @QuLogic for putting it in the example!